### ISSUE TITLE
Issue with using jenssegers/laravel-mongodb package.

### ISSUE THREAD

**Issue by xparthx** [jrean/laravel-user-verification#94]
Hello,

First of all great package, thanks a lot.

When I'm using this package with MongoDB package(https://github.com/jenssegers/laravel-mongodb) I'm getting this error upon signup:

**ErrorException in SerializableClosure.php line 117:
Serialization of closure failed: Serialization of 'MongoDB\BSON\UTCDateTime' is not allowed**

But as soon as I remove code of this package from RegisterController, registration works perfectly.

I have found that this error only generates when we use **sendQueue()** method:

`UserVerification::sendQueue($user, 'E-mail verification');`

This method works fine:

`UserVerification::send($user, 'E-mail verification');`

Any idea how to make it work with MongoDB package?

BTW, I have already checked this, but doesn't helped much:
https://github.com/jenssegers/laravel-mongodb/issues/732

Thanks,
Parth vora

**Comment by jrean** [comment]
Hi,
Thank you.
Can you please share you RegisterController code.

**Comment by xparthx** [comment]
Here is my RegisterController:

```
<?php

namespace App\Http\Controllers\Auth;

use App\User;
use Validator;
use Storage;
use App\Http\Controllers\Controller;
use Illuminate\Foundation\Auth\RegistersUsers;
use App\Http\Controllers\ImageThumbController;
use App\Notifications\SignupComplete;
use Jrean\UserVerification\Traits\VerifiesUsers;
use Jrean\UserVerification\Facades\UserVerification;
use Illuminate\Http\Request;
use Illuminate\Auth\Events\Registered;

class RegisterController extends Controller
{
    use RegistersUsers;
    use VerifiesUsers;
    protected $redirectTo = '/';

    public function __construct()
    {
        $this->middleware('guest', ['getVerification', 'getVerificationError']);
    }

    protected function validator(array $data)
    {
        return Validator::make($data, [
            'name' => 'required|max:255|regex:/^[a-zA-Z\s]*$/',
            'username' => 'required|min:3|max:255|regex:/^[a-zA-Z0-9\s]*$/|unique:users',
            'email' => 'required|email|max:255|unique:users',
            'password' => 'required|min:6|confirmed',
            'password_confirmation' => 'required',
            'contact_no' => 'digits:10',
            'gender' => 'in:male,female',
            'country' => '',
            'hobbies' => '',
            'about_me' => 'min:6',
            'date_of_birth' => 'date',
            'avatar' => 'image',
            ]);
    }

    protected function create(array $data)
    {
        $org_avatar_source = '';
        $avatars_org_dir = 'public/avatars/origional';
        Storage::makeDirectory($avatars_org_dir);

        if (isset($data['avatar'])) {
            if($avatar_file = $data['avatar']->store($avatars_org_dir)) // Upload avatar
            {
                $org_avatar_source = str_replace ('public', 'storage', $avatar_file);
            }
        }

        $user = User::create([
            'name' => $data['name'],
            'username' => $data['username'],
            'email' => $data['email'],
            'password' => bcrypt($data['password']),
            'contact_no' => isset($data['contact_no']) ? $data['contact_no'] : '',
            'gender' => isset($data['gender']) ? $data['gender'] : '',
            'country' => isset($data['country']) ? $data['country'] : '',
            'hobbies' => isset($data['hobbies']) ? $data['hobbies'] : '',
            'about_me' => isset($data['about_me']) ? $data['about_me'] : '',
            'date_of_birth' => isset($data['date_of_birth']) ? $data['date_of_birth'] : '',
            'avatar' => $org_avatar_source,
            'user_type' => 'blogger',
            'social_id' => NULL,
            'registration_type' => 'conventional',
            'deleted_at' => NULL
        ]);  

        if($user) {
            UserVerification::generate($user);
            UserVerification::sendQueue($user, 'E-mail verification');

            $user->notify(new SignupComplete($user));
            return $user;
        }  else {
            return false;
        }
    }
}
```

But I'm pretty sure that it has something to do with **queue** as just **send()** works perfectly, but when we use **sendQueue()** this error comes:

**ErrorException in SerializableClosure.php line 117:
Serialization of closure failed: Serialization of 'MongoDB\BSON\UTCDateTime' is not allowed**

**Comment by jmikola** [comment]
@xparthx: What version of `ext-mongodb` are you using? Note that [1.2.0alpha1](https://pecl.php.net/package/mongodb/1.2.0alpha1) was the first version of the extension to support PHP serialization of the various BSON type classes.

**Comment by xparthx** [comment]
Its 1.1.9. I'm trying to upgrade it to the latest available version. But
there are some issues for that as I'm using windows.

**Comment by jmikola** [comment]
> But there are some issues for that as I'm using windows.

The crash bug that was affecting ZTS users (e.g. Apache on Windows) in versions 1.2.0 and 1.2.1 is resolved in [1.2.2](https://pecl.php.net/package/mongodb/1.2.2), so give that a shot if you haven't already.

**Comment by xparthx** [comment]
Thanks all of you for your time and effort. I have updated **PHP MongoDB extension** from 1.1.9 to 1.2.2 and fixed the bug.
